### PR TITLE
feat: added search, notes to single video & profile page

### PR DESCRIPTION
**Pull request by AbhishekkGautam**
### Search
- Users can filter videos by searching certain keywords.
- mobile responsive.

### Notes to a single video.
- users can write multiple notes while watching a video. 
- users can also delete a particular note.

### Profile page
- users can see their personal information on the profile page. ( first name, last name & email )

### CODE REVIEW

File: src/components/Navbar/Navbar.css [hunk 1]
```diff
@@ -89,6 +89,105 @@
...
+  color: var(--gray-100);
+}
+
+.search-modal-bg {
+  height: 100vh;
+  position: fixed;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
```
Comment (msdeshmukh009) [inline src/components/Navbar/Navbar.css:181]: You can also write `inset:0 0 0 0;`

File: src/components/Navbar/Navbar.js [hunk 2]
```diff
@@ -82,6 +105,47 @@ export const Navbar = () => {
...
+          <div className="">
+            <form
+              className="search-mobile nav-form"
+              onSubmit={e => e.preventDefault()}
+            >
+              <input
+                type="search"
+                placeholder="search..."
+                className="search-input"
+                value={searchQuery || ""}
```
Comment (msdeshmukh009) [inline src/components/Navbar/Navbar.js:133]: The initial value of search query is `" "` so it won't be undefind I guess.

File: src/pages/Home/Home.js [hunk 3]
```diff
@@ -20,12 +20,13 @@ export const Home = () => {
...
             <button
-              onClick={() =>
+              onClick={() => {
                 dispatch({
                   type: "FILTER_BY_CATEGORY",
                   payload: uniqueCategory,
-                })
-              }
+                });
+                dispatch({ type: "FILTER_BY_SEARCH", payload: "" });
```
Comment (msdeshmukh009) [inline src/pages/Home/Home.js]: Its better to extract this as seperate function.

**Review by msdeshmukh009 — APPROVED** [review]
The rest is looking good. One suggestion you can show some message when the user doesn't find any relevant search result.

File: src/components/Navbar/Navbar.js (hunk 2 above)
Comment (AbhishekkGautam) [reply src/components/Navbar/Navbar.js:133]: i did this to avoid warning related to uncontrolled input

File: src/pages/Home/Home.js (hunk 3 above)
Comment (AbhishekkGautam) [reply src/pages/Home/Home.js]: done

**Review by pritam-kr — APPROVED** [review]
Hey, As per my knowledge, your code looks clean, UI and UX are perfect.

**Pull request merged**